### PR TITLE
oc project on kube

### PR DESCRIPTION
**Pull request by deads2k**
Make `oc project` work on kube, so I can avoid having to use the `kubectl config` commands.

### CODE REVIEW

**Comment by fabianofranz** [comment]
@deads2k are you going to close the `--token` one?

**Comment by deads2k** [comment]
> @deads2k are you going to close the --token one?

I can if you want.  I want both commands eventually.  Don't really care how.

**Comment by fabianofranz** [comment]
@deads2k nah, I'm just asking because the `--token` commit is here too.

**Comment by fabianofranz** [comment]
@openshift/cli-review

**Comment by juanvallejo** [comment]
Would this partially address https://github.com/openshift/origin/issues/10821?

**Comment by deads2k** [comment]
Test added.

@stevekuznetsov can we add `test/extended/cmd.sh` to the conformance bucket

File: pkg/cmd/cli/cmd/login/helpers.go [hunk 1]
```diff
@@ -108,8 +110,25 @@ func getHostPort(hostURL string) (string, string, *url.URL, error) {
 	return host, port, parsedURL, err
 }
 
-func whoAmI(client *client.Client) (*api.User, error) {
+func whoAmI(clientConfig *restclient.Config) (*api.User, error) {
+	client, err := client.New(clientConfig)
+
 	me, err := client.Users().Get("~")
+
+	// if we're talking to kube (or likely talking to kube),
```
Comment (juanvallejo) [inline pkg/cmd/cli/cmd/login/helpers.go]: nit: this comment seems a bit confusing

File: pkg/cmd/cli/cmd/login/loginoptions.go [hunk 2]
```diff
@@ -270,6 +263,12 @@ func (o *LoginOptions) gatherProjectInfo() error {
 	}
 
 	projectsList, err := oClient.Projects().List(kapi.ListOptions{})
+	// if we're running on kube (or likely kube), just set it to "default"
+	if kerrors.IsNotFound(err) {
+		fmt.Fprintf(o.Out, "No ACL filtered namespaces available.  Using \"default\".  You switch projects with '%s project <projectname>':\n\n", o.CommandName)
```
Comment (juanvallejo) [inline pkg/cmd/cli/cmd/login/loginoptions.go]: "You can switch projects with..."

**Comment by juanvallejo** [comment]
A few nits, but LGTM

File: pkg/cmd/cli/cmd/login/helpers.go (hunk 1 above)
Comment (deads2k) [inline pkg/cmd/cli/cmd/login/helpers.go]: > nit: this comment seems a bit confusing

rewording suggestion?  Removal?

Comment (ncdc) [inline pkg/cmd/cli/cmd/login/helpers.go]: It makes sense to me. I'm curious what's confusing?

File: pkg/cmd/cli/cmd/login/loginoptions.go (hunk 2 above)
Comment (ncdc) [inline pkg/cmd/cli/cmd/login/loginoptions.go]: "No ACL filtered namespaces available" - will a user have any idea what that means?

Comment (deads2k) [inline pkg/cmd/cli/cmd/login/loginoptions.go]: > "No ACL filtered namespaces available" - will a user have any idea what that means?

I struggled with phrasing.  A kube user wouldn't recognized "projects" either.  Say nothing about it maybe?

Comment (ncdc) [inline pkg/cmd/cli/cmd/login/loginoptions.go]: Yeah, I would remove it if we can't find better wording.

**Comment by stevekuznetsov** [comment]
> can we add `test/extended/cmd.sh` to the conformance bucket

You can append it to the list of tasks in `test/extended/conformance.sh`, sure, if we agree that the tests in `cmd.sh` are in that bucket.

**Comment by deads2k** [comment]
> You can append it to the list of tasks in test/extended/conformance.sh, sure, if we agree that the tests in cmd.sh are in that bucket.

@smarterclayton I plan to do this unless you shout.

**Comment by smarterclayton** [comment]
Hrm - I really do prefer for conformance to be a single suite binary.  We've never really formally solved cmd.sh, so I am a tiny bit worried about that.

**Comment by deads2k** [comment]
> Hrm - I really do prefer for conformance to be a single suite binary. We've never really formally solved cmd.sh, so I am a tiny bit worried about that.

I don't care about the name as much as I care about running it in a test job.

**Comment by fabianofranz** [comment]
[test]

File: pkg/cmd/cli/cmd/login/helpers.go (hunk 1 above)
Comment (juanvallejo) [inline pkg/cmd/cli/cmd/login/helpers.go]: Ah, I had read the comment as if the wording inside the parenthesis was redundant, or maybe accidentally left there after rephrasing the entire sentence, but it made more sense after reading [this comment](https://github.com/openshift/origin/pull/11120/files#diff-823ab4e35468c6cb31f1c30d0e3c41a2R266). Please ignore my first comment, it's fine the way it is

**Comment by stevekuznetsov** [comment]
> I don't care about the name as much as I care about running it in a test job.

Yeah, I guess `conformance` as a term has weight re: what we ship to customers... maybe a reconfig of the conformance job should happen?

**Comment by deads2k** [comment]
comments addressed.  test in `test/extended/cmd.sh`.  I'll leave to someone else to decide how to tie it in, but I suspect we'll want a larger bucket testing commands against kube servers ala `test-cmd.sh`.

**Comment by smarterclayton** [comment]
LGTM

**Comment by deads2k** [comment]
[merge]

**Pull request merged**